src/utilities.js

```diff
@@ -31,6 +35,8 @@ export var $extend = function (destination) {
       if (source[property] && $isplainobject(source[property])) {
         if (!destination.hasOwnProperty(property)) destination[property] = {}
         $extend(destination[property], source[property])
+      } else if (Array.isArray(source[property])) {
+        destination[property] = deepcopy(source[property])
       } else {
         destination[property] = source[property]
       }
```

Comment:
`deepcopy ` method deepcopy the `Object` and `Array`, while the `$extend({}, target)`  can deepcopy an `Object`.